apps/aecore/src/aec_chain_state.erl

```diff
@@ -267,6 +268,15 @@ get_missing_block_hashes(?assert_state() = State) ->
     TopBlockHash  = get_top_block_hash(State),
     get_missing_block_hashes(TopHeaderHash, TopBlockHash, State).
 
+get_top_N_blocks_time_summary(?assert_state() = State, N) ->
```

Comment:
I suggest: is_integer(N), N > 0
to avoid looping infinitely below.

Reply:
@gorillainduction Done. Please revisit.